### PR TITLE
Memory pair game

### PR DESCRIPTION
**Pull request by kt-std**
### Memory Pair Game

[Demo](https://kt-std.github.io/memory-pair-game/)
[Code base](https://github.com/kt-std/memory-pair-game)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/kt-std/memory-pair-game/js/script.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
+const container = document.querySelector(".container"),
+	numCards = 9,
+	cardsData = ((assetsPath) => {
+		const arr = [];
```
Comment (AMashoshyna) [inline submissions/kt-std/memory-pair-game/js/script.js:4]: `arr` is a very generic name. Let's find a name that describes what this variable does.

File: submissions/kt-std/memory-pair-game/js/script.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+		for (let i = 1; i <= numCards; i++) {
```
Comment (AMashoshyna) [inline submissions/kt-std/memory-pair-game/js/script.js:5]: let's try to use `Array.prototype.map` here

File: submissions/kt-std/memory-pair-game/js/script.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+	})("assets/");
+
+let checkedCards = [],
+	checkedCardsContainers = [],
+	shuffledCards = shuffleArray(cardsData),
+	foundMatches = [];
+
+function shuffleArray(o) {
+	for (
+		var j, x, i = o.length;
```
Comment (AMashoshyna) [inline submissions/kt-std/memory-pair-game/js/script.js:19]: why `var`?

File: submissions/kt-std/memory-pair-game/js/script.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+		card.style.width = "0";
+		card.style.height = "0";
+		card.childNodes.forEach(
+			(cardInnerStyle) => (cardInnerStyle.style.display = "none")
+		);
+	}, 500);
+}
+
+function setOpacity(property, value, delay) {
+	if (delay === undefined) {
```
Comment (AMashoshyna) [inline submissions/kt-std/memory-pair-game/js/script.js:107]: I would return the `return` value. Mutating function parameter is really confusing.

File: submissions/kt-std/memory-pair-game/js/script.js [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+	winMessage.appendChild(heading);
+	winMessage.appendChild(button);
+	winMessage.classList.add("win__container", "flex-center");
+	setOpacity(winMessage, 1);
+	document.body.appendChild(winMessage);
+}
+
+function reset() {
+	const clearVars = [checkedCards, checkedCardsContainers, foundMatches].map(
+		(variable) => (variable = [])
```
Comment (AMashoshyna) [inline submissions/kt-std/memory-pair-game/js/script.js:137]: this looks weird. what does it do?

**Review by AMashoshyna — COMMENTED** [review]
Good job! Please take a look at my comments

File: submissions/kt-std/memory-pair-game/js/script.js (hunk 1 above)
Comment (kt-std) [reply submissions/kt-std/memory-pair-game/js/script.js]: @AMashoshyna lots of thanks for your review especially during holidays season! I am so thankful!
I've updated the variable name with another one to clearly express what it actually does.

File: submissions/kt-std/memory-pair-game/js/script.js (hunk 2 above)
Comment (kt-std) [reply submissions/kt-std/memory-pair-game/js/script.js:5]: I’m afraid I’m a little confused how to do this as this function just creates a cardData array that contains "generated" names and pathes.

File: submissions/kt-std/memory-pair-game/js/script.js (hunk 3 above)
Comment (kt-std) [reply submissions/kt-std/memory-pair-game/js/script.js]: `var` is not needed, so I've updated this one with the `let`

File: submissions/kt-std/memory-pair-game/js/script.js (hunk 5 above)
Comment (kt-std) [reply submissions/kt-std/memory-pair-game/js/script.js:137]: This function clears arrays that were used during the game played after restart button was clicked.

File: submissions/kt-std/memory-pair-game/js/script.js (hunk 4 above)
Comment (kt-std) [reply submissions/kt-std/memory-pair-game/js/script.js:107]: Do we actually need to return it? This if statement is only used to set the default value of the delay if this parameter was omitted during the function call.

File: submissions/kt-std/memory-pair-game/js/script.js (hunk 2 above)
Comment (kt-std) [reply submissions/kt-std/memory-pair-game/js/script.js:5]: Would it be okay if I create an array of empty slots with dedicated length and then apply map to it to iterate through all its elements and return the new one?

Comment (kt-std) [reply submissions/kt-std/memory-pair-game/js/script.js]: I've just updated the code with the map method. I do hope it is better now.

File: submissions/kt-std/memory-pair-game/js/script.js (hunk 4 above)
Comment (AMashoshyna) [reply submissions/kt-std/memory-pair-game/js/script.js:102]: I think you are looking for [default parameter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters)

File: submissions/kt-std/memory-pair-game/js/script.js [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+	);
+	[checkedCards, checkedCardsContainers, foundMatches] = clearVars;
```
Comment (AMashoshyna) [inline submissions/kt-std/memory-pair-game/js/script.js:134]: Why do you need these variables?

Comment (AMashoshyna) [reply submissions/kt-std/memory-pair-game/js/script.js:134]: If you don't, then I would suggest to reset the arrays in a simpler manner, like so:
```javascript
checkedCards = []
checkedCardsContainers = []
foundMatches = []
```

File: submissions/kt-std/memory-pair-game/js/script.js (hunk 4 above)
Comment (kt-std) [reply submissions/kt-std/memory-pair-game/js/script.js]: @AMashoshyna, thank you for your time and helpful advises ! I've fixed it with the default parameter.

File: submissions/kt-std/memory-pair-game/js/script.js (hunk 6 above)
Comment (kt-std) [reply submissions/kt-std/memory-pair-game/js/script.js]: I've updated the code with simpler manner of resetting arrays.

**Comment by kt-std** [comment]
@AMashoshyna, I am so sorry for disturbing you :bow:  If there is even a tiny possibility that you will have some free time, I would be unbelievably thankful for checking this PR
cat

**Comment by AMashoshyna** [comment]
Sooo sorry! My bad that the review took so long. Well done!

**Pull request merged**